tensorflow_model_optimization/python/core/sparsity/keras/pruning_impl.py

```diff
@@ -99,9 +124,31 @@ def _update_mask(self, weights):
           tf.math.greater_equal(abs_weights, current_threshold), weights.dtype)
     return current_threshold, new_mask
 
+  def _update_mask_sparsity_2x4(self, weights):
+    """Updates the 2x4 sparsity mask for a given weight tensor.
+
+    This function creates a mask for the given weight tensor so
+    that two elements with the lowest absolute values in the block
+    of four elements are set to be zero. We don't return any threshold.
+
+    Args:
+      weights: The weight tensor that needs to be masked.
+
+    Returns:
+      new_mask: A numpy array of the same size and shape as weights containing
+        0 or 1 to indicate which of the values in weights should be set to zero.
+        If the returned value is None, the requested mask cannot be created.
+    """
+    window_shape = (1, 4)
+    k = 2
```

Comment:
It would be great if these are configurable from the API level, as Nvidia also proposed 1:2 sparsity in the same [paper](https://arxiv.org/pdf/2104.08378.pdf).

Reply:
Hi @liyunlu0618 Thanks for the review. Yes, we can add 1:2 in addition. We will do this.